src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+or a string for an object source) 
+* Returns: none
+
+Iterates over a supplied array or hash, invoking a function once for each element. Similar to [`jQuery.each`](http://api.jquery.com/jquery.each/) only the
+arguments to `func` are the right way round and the function does not explode on nonvalues. `fluid.each` on `null` or `undefined` is a no-op.
+
+### fluid.invokeLater(func) 
+
+A standard utility to schedule the invocation of a function after the current function call stack returns. On browsers this defaults to `setTimeout(func, 1)` but in
+other environments can be customised - e.g. to `process.nextTick` in node.js. In future, this could be optimised in the browser to not dispatch into the event queue
+
+* `func {Function}` A function accepting zero arguments which is to be invoked later
+* Returns: none
+
+### fluid.allocateGuid()
+
+Allocate a string value that will be unique within this Infusion instance (frame or process), and
+globally unique with high probability (50% chance of collision after a million trials). This has the structure `prefix-id` where
+`prefix` is the value of `fluid.fluidInstance` identifying this Infusion instance, and `id` is a increasing integer initialised at
+1 when this infusion instance starts up.
+
+* Returns: `{String}` A string of about a dozen characters that has made moderate efforts to be globally unique.
+
+
+## Creation, copying and destroying
+
+### fluid.freshContainer(tocopy)
+
+* `tocopy {Object|Array}`
+* Returns: `{Object|Array}`
+
+Return an empty container as the same type as the argument (either an array or hash). To be wise, the supplied argument should 
+pass the `fluid.isPlainObject` test.
+
+### fluid.copy(tocopy)
+
+* `tocopy {Any}`
+* Returns: `{Any}`
+
+Performs a deep copy (clone) of its argument. This will guard against cloning a circular object by terminating if it reaches a path depth
+greater than `fluid.strategyRecursionBailout`. `fluid.copy` will not copy an object which passes `fluid.isUncopyable`.
+
+### fluid.makeArray(arg)
+
+* `arg {Any}`
+* Returns: `{Array}`
+
+Converts the supplied argument to an array (or copies it if it is already an array), by a variety of strategies:
+
+* If it is even slightly arrayable (has a numeric `length` property), its elements will be copied into a fresh array
+* If it is not slightly arrayable, it will be placed into a 1-element array
+* If it is undefined, a 0-element array will be returned
+
+### fluid.generate(n, generator[, applyFunc])
+
+Fills an array of given size with copies of a value or result of a function invocation
+* `n {Integer}` The size of the array to be filled
+* `generator {Object|Function}` Either a value to be replicated or function to be called. In case of a function, the signature is `(i {Integer}) → Any`, accepting the index position and returning the generated element.
+* `applyFunc {Boolean}` If `true`, the `generator` value is to be treated as a function
```

Comment:
You should clarify that if applyFunc is false, then generator will just be copied, even if it is a function. And I ask: if generator is a function and applyFunc is false, am I still bound by the signature described above? And does applyFunc have a default value?